[WebGPU] copyTextureToTexture does not work with compressed formats

mwyrzykowski opened the pull request:
#### f6ed3474524d1341c873608922814812d0b5fcda
<pre>
[WebGPU] copyTextureToTexture does not work with compressed formats
<a href="https://bugs.webkit.org/show_bug.cgi?id=254103">https://bugs.webkit.org/show_bug.cgi?id=254103</a>
&lt;radar://106887775&gt;

Reviewed by Tadeu Zagallo.

Texture::isValidImageCopySource / isValidImageCopyDestination are only supposed
to be called on depth or stencil formats. They do not return expected values
except for depth formats.

* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::validateWriteTexture):
Only validate if this is a depth format.

* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::isValidDepthStencilCopySource):
(WebGPU::Texture::isValidDepthStencilCopyDestination):
Assert this is a depth or stencil format.

Canonical link: <a href="https://commits.webkit.org/261880@main">https://commits.webkit.org/261880@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e3cca43b7c2b19f90dedd3e77733034d7f4d3270

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/113092 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/22225 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/90/builds/1762 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/4863 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/121555 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/23595 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/13408 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/6106 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/118879 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/17533 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/100796 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/106174 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/99487 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/88/builds/1332 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/46561 "38 flakes 109 failures 1 missing results") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/14527 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/89/builds/1374 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/15242 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/10708 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/20536 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/53366 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/8296 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/17088 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebGPU/WebGPU/Texture.mm:
```diff
@@ -2584,10 +2585,10 @@ static WGPUExtent3D imageCopyTextureSubresourceSize(const WGPUImageCopyTexture&
 bool Texture::isValidImageCopyDestination(WGPUTextureFormat format, WGPUTextureAspect aspect)
```
If this only deals with depth/stencil formats, I feel like this function has the wrong name; Image feels to generic.

mwyrzykowski replied:
I agree, maybe `isValidDepthStencilCopyDestination` and `isValidDepthStencilCopySource`?

djg reviewed (commented):
:shipit:

mwyrzykowski committed:
[WebGPU] copyTextureToTexture does not work with compressed formats
https://bugs.webkit.org/show_bug.cgi?id=254103
<radar://106887775>

Reviewed by Tadeu Zagallo.

Texture::isValidImageCopySource / isValidImageCopyDestination are only supposed
to be called on depth or stencil formats. They do not return expected values
except for depth formats.

* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::validateWriteTexture):
Only validate if this is a depth format.

* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::isValidDepthStencilCopySource):
(WebGPU::Texture::isValidDepthStencilCopyDestination):
Assert this is a depth or stencil format.

Canonical link: https://commits.webkit.org/261880@main
